shared/hooks/useDateFilter.ts

```diff
@@ -0,0 +1,61 @@
+import { DateTime } from 'luxon'
+import { useState, useEffect } from 'react'
+
+import { EventModel } from 'shared/api'
+
+type UseDateFilterReturn = {
+  fromDate: DateTime | null
+  setFromDate: (fromDate: DateTime | null) => void
+  toDate: DateTime | null
+  setToDate: (toDate: DateTime | null) => void
+  filteredEvents: EventModel[]
+  fromDateError: string | null
+  toDateError: string | null
+}
+
+const useDateFilter = (events: EventModel[] | null): UseDateFilterReturn => {
+  const [fromDate, setFromDate] = useState<DateTime | null>(DateTime.now())
+  const [toDate, setToDate] = useState<DateTime | null>(DateTime.now().plus({ day: 10 }))
+  const [filteredEvents, setFilteredEvents] = useState<EventModel[]>([])
+  const [fromDateError, setFromDateError] = useState<string | null>(null)
+  const [toDateError, setToDateError] = useState<string | null>(null)
+  useEffect(() => {
+    const filterByDateRange = (from: DateTime | null, to: DateTime | null) => {
+      setToDateError('')
+      setFromDateError('')
+
+      if (!from) {
+        setFromDateError('invalidFromDate')
+        return []
+      }
+      if (!to) {
+        setToDateError('invalidToDate')
+        return []
+      }
+
+      const fromDateTime = from
+      const toDateTime = to.endOf('day')
+
+      if (fromDateTime > toDateTime) {
+        setFromDateError('shouldBeEarlier')
+        return []
+      }
+      return events?.filter(event => event.date.startDate >= fromDateTime && event.date.endDate <= toDateTime) || []
```

Comment:
I think this condition is not entirely correct, it excludes events where a day overlaps, i.e. the filter is set from
2024-09-13 to 2024-09-22 and the event is from 2024-09-12 to 2024-09-13. With this condition, it is filtered out even though it should be shown (since it happens on 13th).
The following condition should solve this:
```suggestion
      return events?.filter(event => event.date.startDate <= toDateTime && event.date.endDate >= fromDateTime) || []
```